core/src/node_id.rs

```diff
@@ -0,0 +1,232 @@
+use std::fmt::Display;
+use std::{fs, path::PathBuf, str::FromStr};
+
+use crate::{StreamId, StreamIdType};
+use anyhow::{anyhow, Context, Ok, Result};
+use cid::multihash::Multihash;
+use cid::Cid;
+use libp2p_identity::PeerId;
+use ring::signature::{Ed25519KeyPair, KeyPair};
+
+/// NodeId is the public_ed25519_key_bytes of the node
+#[derive(Clone, Eq, PartialEq, Default, Copy)]
+pub struct NodeId {
```

Comment:
I find default a bit weird on this but it does seem useful in the tests. 

Below I suggested we should stop using references and just pass an owned version of this because it's Copy.. but now I'm wondering if this should be copy. We call `did_key()` for every insert now, which means doing the multibase encoding and allocating a string.. should we track more than 32 bytes of public key on this? Does it make sense to keep the a DidDocument (or at least a did:key string) on self? 

I like that we have a NodeId, but it seems a bit odd to have _another_ new construct when we already have DIDs and PeerIds and it'd be nice to consolidate.. I don't feel super strongly but I'm not sure when I should pick which one as they're all just variations of public keys.